### PR TITLE
Add prod as a protected environment.

### PR DESCRIPTION
**Pull request by benguillet**

### CODE REVIEW

**Comment by sanjayprabhu** [comment]
Yesssss! Thank you @benguillet 

:+1:

File: deploy.go [hunk 1]
```diff
@@ -75,8 +75,9 @@ var flags = []cli.Flag{
 	},
 }
 
-var ProtectedEnvironments = map[string]bool{
-	"production": true,
+var ProtectedEnvironments = map[string]struct{}{
+	"production": struct{}{},
```
Comment (ejholmes) [inline deploy.go]: Not a big deal, but think I liked `map[string]bool` better.

**Comment by ejholmes** [comment]
:+1:

**Comment by ejholmes** [comment]
I wonder if it would be better to be able to set some aliases that get converted to the real thing. e.g. if you specify `prod` here it turns it into `production` before creating the github deployment. That would address my concern in https://github.com/remind101/empire/pull/681#issuecomment-159134486.

**Comment by benguillet** [comment]
@ejholmes this was my first idea, but since this is open sourced, what about people using this to match to an environment named `prod`? This would break it for them (deploy is versioned, but i feel like we should pass whatever environment the user wants and it's the server's job to support it).

**Comment by ejholmes** [comment]
I was thinking more of making it configurable, like in a config file or an environment variable. Anyways, I don't think we need to block this on that. This method is still a good option and we can always change it later.

**Pull request merged**